xla/service/hlo_computation_deduplicator.cc

```diff
@@ -78,20 +78,42 @@ StatusOr<bool> HloComputationDeduplicator::Run(
     }
     return false;
   };
-  for (HloComputation* comp :
-       module->MakeComputationPostOrder(execution_threads)) {
+  // Let us collect the opcode of the callers of each computation. We will only
+  // deduplicate computations with the same caller opcode.
```

Comment:
Nit: you're describing "what" the code does, but not "why": the latter is in general the most important form of documentation (the "what" can always be recovered from the code).
Can you elaborate on the invariant to preserve?

Also if different callers opcode can't share the same computation, that should be checked in the verifier: can we add this check?

Reply:
As a newbie to XLA, I am on a thin ice when describing pipeline-wide invariants, but let me take a stab at a better comment.